src/components/AnalysisPage/ThicknessAnalysisBox/DecimalInput.jsx

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from "react";
+import { Input } from "antd";
+import PropTypes from "prop-types";
+
+const DecimalInput = ({ value, onChange }) => {
+  const [inputValue, setInputValue] = useState(true);
+
+  useEffect(() => {
+    setInputValue(value?.toString());
+  }, [value]);
+
+  const handleChange = (e) => {
+    const newValue = e.target.value;
+    const reg = /^(\d+)?(\.\d{0,2})?$/;
+
+    if (newValue === "" || reg.test(newValue)) {
+      setInputValue(newValue);
+      const numericValue = newValue === "" ? null : parseFloat(newValue);
+      onChange(numericValue);
+    }
+  };
```

Comment:
Add a comment to this saying that it only allows numbers and limits the the input to 2 decimal spots. I would have no idea what this is doing if it weren't for the commit message lol

Reply:
Also I found a small bug. If you just type `.` as the first value in the input. It will display it as NaN (not a number)
![nan-bug](https://github.com/rknizzle/propermesh-view/assets/40612461/a5515cc0-96f4-4696-be2d-862bab3dc2ee)

Reply:
I briefly saw that bug last night buut couldn't replicate it!!

Nice find, nice find

Reply:
@rknizzle I pushed up a fix for the NaN bug!

seems to be working.

The only thing I don't like is that if you have a value like "0.09" and go to delete just the 9. It changes the value to just "0" instead of "0.0".

Not that big of a deal I don't think. But still.

Also, you can definitely merge this in. 
But if there are too many values in the threshold history row, they will eventually start to leave the container.

<img width="470" alt="image" src="https://github.com/rknizzle/propermesh-view/assets/111093132/0281e9b2-d58d-4ed8-bd28-dcd4185a5cc0">


I can take care of this when I do the responsiveness and layout of the analysis page as a whole. So we can move on if you'd like

Reply:
I actually hate that you can't only delete the second decimal spot.

I'll eventually fix that too,

Reply:
Cool ya lets try to get that fixed at some point but I'm going to merge this in and we can move onto thickness visualization viewing and the layout of the components in the page

I created an 'issue' so we can go back to this later:
https://github.com/rknizzle/propermesh-view/issues/42